xblock/mixins.py

```diff
@@ -244,32 +245,48 @@ def save(self):
         if not self._dirty_fields:
             # nop if _dirty_fields attribute is empty
             return
+
+        fields_to_save = self._get_fields_to_save()
+
+        self.force_save_fields(fields_to_save)
+
+    def force_save_fields(self, field_names):
+        """
+        Save all fields that are specified in `field_names`, even
+        if they are not dirty.
+        """
+        fields = [self.fields[field_name] for field_name in field_names]
+        fields_to_save_json = {}
+        for field in fields:
+            fields_to_save_json[field.name] = field.to_json(self._field_data_cache[field.name])
+
         try:
-            fields_to_save = self._get_fields_to_save()
             # Throws KeyValueMultiSaveError if things go wrong
-            self._field_data.set_many(self, fields_to_save)
-
+            self._field_data.set_many(self, fields_to_save_json)
         except KeyValueMultiSaveError as save_error:
-            saved_fields = [field for field in self._dirty_fields if field.name in save_error.saved_field_names]
+            saved_fields = [field for field in fields if field.name in save_error.saved_field_names]
             for field in saved_fields:
                 # should only find one corresponding field
-                del self._dirty_fields[field]
-            raise XBlockSaveError(saved_fields, self._dirty_fields.keys())
+                fields.remove(field)
+                # if the field was dirty, delete from dirty fields
+                self._reset_dirty_field(field)
+            raise XBlockSaveError(saved_fields, fields)
 
         # Remove all dirty fields, since the save was successful
-        self._clear_dirty_fields()
+        for field in fields:
+            self._reset_dirty_field(field)
 
     def _get_fields_to_save(self):
         """
-        Create dictionary mapping between dirty fields and data cache values.
-        A `field` is an instance of `Field`.
+        Get an xblock's dirty fields.
         """
-        fields_to_save = {}
+        fields_to_save = []
         for field in self._dirty_fields.keys():
             # If the field value isn't the same as the baseline we recorded
             # when it was read, then save it
             if field._is_dirty(self):  # pylint: disable=protected-access
-                fields_to_save[field.name] = field.to_json(self._field_data_cache[field.name])
+                fields_to_save.append(field.name)
+
         return fields_to_save
```

Comment:
Probably doesn't need to happen with this PR, but this could be simplified and inlined as:

    [field.name for field in self._dirty_fields if field.is_dirty(self)]